Workshop notebooks and additional explanations

moose-in-australia opened the pull request:
The main additions of this pull request are:
* Updated notebooks with more explanation between code cells
* New version of notebooks for workshops, where some cells have been emptied to challenge participants
* New notebook which guides a user through cleaning up all the resources produced by the notebooks

chrisking commented:
Feedback:

1. First of all THANK YOU implementing the polling on all solution version creates, I toyed around with a few ideas and never settled on one, happy to see it just delivered.
2. In notebook 2 `<img src="static/imgs/personalize_metrics.png" alt="How Personalize splits data for evaluation" width=800/>` does not render as MD for the image.
3. Can you add the cleanup notebook as just a step 4? It follows linearly, just connect it to the main readme and notebook 3.
4. There's a workshop folder and notebooks in root, is there a reason for the duplication?

moose-in-australia commented:
1. No worries, it could probably be implemented in a more efficient way, but I figured it's a start.
2. Updated the image code to MD instead of HTML. Displays fine now.
3. Sure, I've renamed the clean up notebook to be number 4 in the series, and added it to the workshop and completed folders as well.
4. The workshop folder contains the same notebooks as the root folder, except some of the cells have been emptied to challenge the participant to write code themselves. This is based on feedback that I often get from my workshop participants, where they ask for notebooks which are not 'Run All', but which requires them to do some thinking. I think the main notebooks in the root folder should just be 'Run All' notebooks which provide all the code for someone who just wants to see how to do it, and then I placed the challenge notebooks in a folder called workshop because that's the only time I would use them instead of the main notebooks. Open to other ways of organizing this though.

Also, one question I had - is there a reason why the completed folder contains a copy of everything instead of just containing copies of the notebooks? It seems to me that the YML files and perhaps even the README wouldn't need to be copied in that folder.

chrisking commented:
Merging the pull request, and to answer it was due to lazy copying on why they are all in the completed folded(no better answer).
